services/controllerhost/consumer_test.go

```diff
@@ -140,6 +204,58 @@ func (s *McpSuite) TestCGExtentSelectorHonorsCreatedTime() {
 	}
 }
 
+func (s *McpSuite) TestCGExtentSelectorHonorsCreatedTimeKafka() {
+	dstPath := s.generateName("/test/selector")
+	dstDesc, err := s.createDestination(dstPath, shared.DestinationType_KAFKA)
+	s.Nil(err, "Failed to create destination")
+
+	cgPath := s.generateName("/test/selector-cg")
+	cgDesc, err := s.createConsumerGroup(dstPath, cgPath)
+	s.Nil(err, "Failed to create consumer group")
+
+	hosts, _ := s.mockrpm.GetHosts(common.StoreServiceName)
+	stores := []string{hosts[0].UUID, hosts[1].UUID, hosts[2].UUID}
+	inhosts, _ := s.mockrpm.GetHosts(common.InputServiceName)
+
+	nExtents := 0
+	context := s.mcp.context
+	dstID := dstDesc.GetDestinationUUID()
+	extents := make([]string, 0, 10)
+
+	for i := 0; i < 5; i++ {
+		extID := uuid.New()
+		context.mm.CreateExtent(dstID, extID, inhosts[0].UUID, stores)
+		context.mm.SealExtent(dstID, extID)
+		context.mm.MoveExtent(dstID, dstID, extID, cgDesc.GetConsumerGroupUUID())
+		extents = append(extents, extID)
+		// sleep to advance createdTime for next extent
+		time.Sleep(2 * time.Millisecond)
+		nExtents++
+	}
+
+	cgExtents := newCGExtentsByCategory()
+
+	gotExtents, avail, err := selectNextExtentsToConsumeKafka(context, dstDesc, cgDesc, cgExtents, metrics.GetOutputHostsScope)
+	s.Nil(err, "selectNextExtentsToConsume() error")
+
+	s.Equal(maxExtentsToConsumeForDstKafka, len(gotExtents), "Wrong number of next extents to consume")
+	s.Equal(numKafkaExtentsForDstKafka+nExtents, avail, "Wrong number of available extents")
+
+	var nPhantom, nDlq int
+	for _, x := range gotExtents {
+		if x.GetStoreUUIDs()[0] == kafkaPhantomStoreUUID {
+			nPhantom++
+		} else {
+			s.Equal(extents[nDlq], x.GetExtentUUID(), "Extents not served in time order")
+			nDlq++
+		}
+	}
+
+	// ensure that the number of extents picked are within bounds
+	s.Equal(numKafkaExtentsForDstKafka, nPhantom, "Wrong number of phantom extents")
+	s.Equal(maxDlqExtentsForDstKafka, nDlq, "Wrong number of phantom extents")
+}
+
 func (s *McpSuite) TestCGExtentSelectorHonorsDlqQuota() {
```

Comment:
please add a similar test to make sure we don't make the correct number of dlq extents available for consumption